tools/scripts/Makefile

```diff
@@ -0,0 +1,265 @@
+LIBRARY_DIR 	  = $(BUILD_DIR)/bsp/$(PROCESSOR_CORE)/lib
+PLATFORM_DRIVERS  = $(DRIVERS_DIR)/platform/LOCAL_PLATFORM
+PROJECT_DIR 	  = $(PROJECTS_DIR)/$(PROJECT)
+
+# uncomment this line to include a custom config
+#include $(SCRIPTS_DIR)/config.mk
+include $(PROJECTS_DIR)/$(PROJECT)/src.mk
+
+# Default verbosity level is 0
+VERBOSE ?=0
+ifeq ($(VERBOSE),1)
+MUTE =
+NULL = 
+else
+MUTE = @
+NULL = >/dev/null
+endif
+
+EXEC = release
+TIMESTAMP = $(shell date +"%T")
+ECHO = @echo -n "\\e[32m[$(TIMESTAMP)]\\e[39m"
+ERROR = @echo -n "\\e[32m[$(TIMESTAMP)]\\e[31m"
+CC ?= gcc
+
+# Xilinx platform handling
+ifeq (xilinx,$(strip $(PLATFORM)))
+
+CFLAGS = -Wall 								\
+	 -Wmissing-field-initializers					\
+	 -Wclobbered 							\
+	 -Wempty-body 							\
+	 -Wignored-qualifiers 						\
+	 -Wmissing-parameter-type					\
+	 -Wold-style-declaration					\
+	 -Woverride-init 						\
+	 -Wsign-compare							\
+	 -Wtype-limits							\
+	 -Wuninitialized						\
+	 -Wunused-but-set-parameter					\
+	 -Wno-unused-parameter						\
+	 -fdata-sections						\
+	 -ffunction-sections 						\
+	 -std=gnu90							\
+	 -O2 								\
+	 -g3								\
+	 -MMD 								\
+	 -MP								\
+	 -lm						
+	#-Werror
+
+LDFLAGS = -Wl,-T							\
+	  -Wl,$(BUILD_DIR)/app/src/lscript.ld -L $(LIBRARY_DIR)
+
+# Zynq
+ifeq (ps7_cortexa9_0,$(strip $(PROCESSOR_CORE)))
+
+CC = arm-none-eabi-gcc
+
+CFLAGS += -mcpu=cortex-a9 						\
+	  -mfpu=vfpv3 							\
+	  -mfloat-abi=hard
+
+LDFLAGS += -specs=$(BUILD_DIR)/app/src/Xilinx.spec 			\
+	   -mfpu=vfpv3							\
+ 	   -mfloat-abi=hard 						\
+	   -mcpu=cortex-a9						\
+	   -Wl,-build-id=none
+endif
+# Zynqmp 
+ifeq (psu_cortexa53_0,$(strip $(PROCESSOR_CORE)))
+
+CC = aarch64-none-elf-gcc
+
+endif
+# Microblaze
+ifeq (sys_mb,$(strip $(PROCESSOR_CORE)))
+
+CC = microblaze-xilinx-elf-gcc
+
+CFLAGS += -mcpu=cortex-a9 						\
+	  -DXILINX -DMICROBLAZE						\
+	  -DXILINX_PLATFORM						\
+	  -mlittle-endian						\
+	  -mxl-barrel-shift						\
+	  -mxl-pattern-compare 						\
+	  -mno-xl-soft-div						\
+	  -mcpu=v11.0							\
+	  -mno-xl-soft-mul						\
+	  -mxl-multiply-high
+
+LDFLAGS += -Xlinker --defsym=_HEAP_SIZE=0x100000 			\
+	   -Xlinker --defsym=_STACK_SIZE=0x2000 			\
+	   -mlittle-endian 						\
+	   -mxl-barrel-shift						\
+	   -mxl-pattern-compare						\
+	   -mno-xl-soft-div						\
+	   -mcpu=v11.0							\
+	   -mno-xl-soft-mul						\
+	   -mxl-multiply-high 						\
+	   -Wl,--no-relax 						\
+	   -Wl,--gc-sections 
+
+endif
+
+# Common xilinx libs
+LIBS =	-Wl,--start-group,-lxil,-lgcc,-lc,--end-group
+
+endif
+
+# Add the common include paths
+INC_PATHS = $(foreach x, $(INCS), $(addprefix -I ,$(dir $(x))))
+LIB_PATHS = $(foreach x, $(LIBRARY_DIR), $(addprefix -L ,$(x)))
+EXTRA_INCS =
+
+OBJS = $(SRCS:.c=.o)
+
+# Compile all the source files
+%.o : %.c
+	$(ECHO) [CC]	$(notdir $<) '\n'
+	$(MUTE)$(CC) $(CFLAGS) $(INC_PATHS) $(EXTRA_INCS)		\
+	-c $(addprefix $(BUILD_DIR)/app/src/,$(notdir $<))		\
```

Comment:
curios: why `$(addprefix)` ?
this looks a bit complicated;

usually, it's a good idea for OBJS to be defined with full paths

Reply:
While building the project, one of the steps is to copy the srcs from the no-Os repo to a local srcs folder. With `$(addprefix)` and `$(notdir)` I redefine [the path of] the compiled sources to be the local ones, and not the repository's. Also, while generating the object files, I prefer to save them in a separate folder called _obj_.